### PR TITLE
sub / settings menu haptics update

### PR DESCRIPTION
**Pull request by stella3d**
This adds haptic feedback to all the Toggles that are on the main menu that did not have them. 

In the process, i made a base class for the menu toggles and buttons.  The events aren't in the base class because working with inherited events proved to not work well.  Toggles have simpler hover actions since they don't generally need / have tooltips.

I used a weaker click pulse, because the longer pulse was too long for toggles, it felt like.  this one feels good to me.

I also had to update the LocomotionTool's settings UI to use TextMeshPro.

### CODE REVIEW

**Comment by stella3d** [comment]
this should resolve #383

File: Menus/MainMenu/Scripts/MainMenuToggle.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+        new void Awake()
+        {
+            m_Selectable = m_Toggle;
+            m_OriginalColor = m_Toggle.targetGraphic.color;
+        }
+
+        public void OnRayEnter(RayEventData eventData)
+        {
+#if INCLUDE_TEXT_MESH_PRO
```
Comment (AndrewTHEManeri) [inline Menus/MainMenu/Scripts/MainMenuToggle.cs]: Don't think this is #if block is needed?

File: Menus/MainMenu/Scripts/MainMenuSelectable.cs [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+        [SerializeField]
+        HapticPulse m_HoverPulse;
+
+#if INCLUDE_TEXT_MESH_PRO
+        [SerializeField]
+        protected TextMeshProUGUI m_Description;
+#endif
+
+#if INCLUDE_TEXT_MESH_PRO
```
Comment (mtschoen-unity) [inline Menus/MainMenu/Scripts/MainMenuSelectable.cs]: Group these fields in the same #ifdef. I know this was my code but I did it wrong :)

File: Menus/MainMenu/Scripts/MainMenuToggle.cs (hunk 1 above)
Comment (mtschoen-unity) [reply Menus/MainMenu/Scripts/MainMenuToggle.cs]: Yeah, there are no TMP references in this entire class. Looks like it's left over from before the refactor to MainMenuSelectable

**Comment by stella3d** [comment]
think that should handle updating to merge with latest dev & PR feedback.  removed un-needed if defs.

**Comment by dunity** [comment]
There appear to be conflicts with the SettingsUI prefab & this branch.  Please be careful not to overwrite any recent fixes/changes on that prefab when merging latest development.

**Comment by mtschoen-unity** [comment]
@dunity I merged in dev, updated TMP on the settings UI that needed it, disabled hovers on non-interactable MenuButtons and MenuToggles, and added hovers to Toggles that aren't on. Basically we check interactable, rather than isOn to block hovers.

I also added a MainMenuButton to the "breadcrumb" that says "Snapping" so that it sends hover/click pulses.

Also deleted unused pulses from MainMenuButton.

Approved on my end. Waiting on @dunity's approval to merge do dev.

**Comment by mtschoen-unity** [comment]
Manual merge successful. Performing Github merge.

**Pull request merged**